src/query.rs

```diff
@@ -183,8 +190,42 @@ pub struct QueryOverride {
     pub lint_level: Option<LintLevel>,
 }
 
-/// A mapping of lint ids to configured values that override that lint's defaults.
-pub type OverrideMap = BTreeMap<String, QueryOverride>;
+/// A mapping of `Identifier`s to configured values that override those lints' defaults.
+pub type OverrideMap = BTreeMap<Identifier, QueryOverride>;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum LintGroup {
+    MustUseAdded,
+}
+
+/// An identifier that matches a lint, or a group of lints.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum Identifier {
+    /// Identifies a group of lints, based on the [`SemverQuery::lint_group`] field.
+    Group(LintGroup),
+    /// Identifies a singular lint by id.
+    Lint(String),
+}
+
+impl Identifier {
+    /// Returns a list of lint ids that this identifier refers to, given the set of
+    /// all queries.
+    pub(crate) fn lints<'a>(
+        &'a self,
+        all_queries: &'a BTreeMap<String, SemverQuery>,
+    ) -> Vec<&'a str> {
```

Comment:
We could also make this a `Vec<&'a SemverQuery>`, but we would need to handle the case where an unknown lint id was provided (but this may be a good thing, currently there is no warning for this)

Reply:
`Identifier` doesn't quite strike me as the right place for this logic. We're giving the larger pool of data (`&'a BTreeMap<String, SemverQuery>`) to the smaller type (a two-variant enum) and asking it to figure it out.

Also, I feel like we want the "resolve the identifiers" logic to have `Result` types so we can complain about bad IDs more explicitly. And we really need to test those cases before someone gets bitten by a silent bug around bad IDs...